### PR TITLE
PG JobStore

### PR DESCRIPTION
**Pull request by TMaYaD**
This fork adds PG Jobstore which saves the jobgraph data and file list to postgres and saves the actual files to s3 or local filesystem.

Jobstore url should be of the form `PG_DATABASE_URL+FILESTORE_URL'
ex: `postgresql://testuser:testpass@localhost:5432/toil#namespace+s3://lx-test-file-store/namespace`

TODO: Add support for namespacing the PG entries so that same database can be used from multiple instanced by assigning different namespaces.

### CODE REVIEW

File: src/toil/jobStores/pg/jobStore.py [hunk 1]
```diff
@@ -0,0 +1,548 @@
...
+        job = JobGraph.fromJobNode(jobNode, jobStoreID=self._newJobID(), tryCount=self._defaultTryCount())
+        try:
+            with self.conn.cursor() as cur:
+                attrs = job.__dict__
+                attrs['predecessorsFinished'] = list(attrs['predecessorsFinished'])
+                attrs.pop('_config')
+                self._insert_row('job_store', **attrs)
+                self.conn.commit()
+        except RuntimeError as e:
+            # Handle known erros
```
Comment (ramyala) [inline src/toil/jobStores/pg/jobStore.py]: known errors

File: src/toil/jobStores/pg/jobStore.py [hunk 2]
```diff
@@ -0,0 +1,548 @@
...
+                    job.logJobStoreFileID,
+                    job.checkpoint,
+                    job.checkpointFilesToDelete,
+                    cPickle.dumps(job.chainedJobs),
+                    self.namespace,
+                    job.jobStoreID,
+                ))
+            self.conn.commit()
+        except RuntimeError as e:
+            # Handle known erros
```
Comment (ramyala) [inline src/toil/jobStores/pg/jobStore.py]: looks like similar typo here and below too

File: src/toil/jobStores/pg/jobStore.py [hunk 3]
```diff
@@ -0,0 +1,548 @@
...
+            raise e
+
+    # NOTE: The embedded PL/pgSQL script is playing havoc with atom syntax highlighter.
+    #       Keep this function at the end to limit the pain.
+    def _create_sql_triggers(self):
+        try:
+            with self.conn.cursor() as cur:
+                function_plsql = """
```
Comment (ramyala) [inline src/toil/jobStores/pg/jobStore.py]: heredoc() might help here for formatting errors, etc.

Comment (TMaYaD) [inline src/toil/jobStores/pg/jobStore.py]: Could you point me to the reference? I couldn't find any by googling. 
The string is valid heredoc and works fine as such. It's only the atom's syntax parser that goes haywire. vi's works fine.

**Pull request merged**